### DISCUSSION TITLE
Expression Column and Filters

### DISCUSSION THREAD

**Discussion by yangxuesong** [perspective-dev/perspective#2488] [Q&A]
I want to use indexof function in expression column,
e.g. 
var v[2] := {-1,-1};
if(indexof("SomeColumn", 'SomePattern', v) OR indexof(...)) {'contains'} else {'not contains'}

BUT not work as expected, can anyone provide a good example of how to use indexof function and OR operator to implement `string.contains` function? Thanks in advance!
BTW, if I set 'SomePattern' to some unicode string, there will be an error

Can I use OR between filters?

**Comment by texodus** [comment]
This is poorly documented for sure - we're working on adding full docs for this.

The issue here is that the regex engine we use has a unique interpretation of capture grouping, and will _only_ match on a capture group (not a raw string). So just put your regex match in a capture group and it should work, e.g. use `'(SomePattern)'` instead of `'SomePattern'`:

```
var v[2] := {-1,-1};
if (indexof("SomeColumn", '(SomePattern)', v) OR indexof(...)) {
    'contains'
} else {
    'not contains'
}
```

**Comment by yangxuesong** [comment]
Thanks for your reply!
I need to use unicode chars in the '(SomePattern)', e.g '(北京)',  there will
be an error: Invalid string token: (

Andrew Stein ***@***.***> 于2024年1月18日周四 04:37写道：

**Reply by oleksify** [reply]
@texodus We're having same issue with unicode chart in expression columns (Ukrainian language, Cyrillic). We have to do some tricks in js to translate columns in runtime. I wonder if I maybe missing something or it's intentional behavior/limitation. Thanks!
